Formula/perl.rb

```diff
@@ -25,26 +26,34 @@ class Perl < Formula
   depends_on "gdbm"
 
   uses_from_macos "expat"
+  uses_from_macos "libxcrypt"
+
+  on_linux do
+    # Configure checks if libnsl is present but does not link to it.
+    # Disabling the libnsl check is not possible without hacky solutions that are prone
+    # to bitrot.  Remove this if upstream provides a more reliable way to skip this check.
+    depends_on "libnsl" => :build
+  end
 
   # Prevent site_perl directories from being removed
   skip_clean "lib/perl5/site_perl"
 
   def install
     args = %W[
       -des
-      -Dprefix=#{prefix}
-      -Dprivlib=#{lib}/perl5/#{version}
-      -Dsitelib=#{lib}/perl5/site_perl/#{version}
-      -Dotherlibdirs=#{HOMEBREW_PREFIX}/lib/perl5/site_perl/#{version}
+      -Dinstallprefix=#{prefix}
+      -Dprefix=#{opt_prefix}
+      -Dprivlib=#{opt_lib}/perl5/#{version.major_minor}
+      -Dsitelib=#{opt_lib}/perl5/site_perl/#{version.major_minor}
+      -Dotherlibdirs=#{HOMEBREW_PREFIX}/lib/perl5/site_perl/#{version.major_minor}
       -Dperlpath=#{opt_bin}/perl
       -Dstartperl=#!#{opt_bin}/perl
-      -Dman1dir=#{man1}
-      -Dman3dir=#{man3}
+      -Dman1dir=#{opt_share}/man/man1
+      -Dman3dir=#{opt_share}/man/man3
       -Duseshrplib
       -Duselargefiles
       -Dusethreads
```

Comment:
@cho-m, it'd be useful if you could take a look at this since you tried doing something similar before.